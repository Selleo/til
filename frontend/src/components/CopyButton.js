import React from 'react'
import { ReactComponent as Share } from '../assets/icons/share.svg'

const CopyButton = props => {
  const { isInTooltip, handleClick, text } = props

  return (
    <button onClick={handleClick} className="post__share-button">
      {!isInTooltip && <Share />}
      <span className="post__share-text">{text}</span>
    </button>
  )
}

export default CopyButton
