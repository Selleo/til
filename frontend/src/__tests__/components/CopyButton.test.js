import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import CopyButton from '../../components/CopyButton'

describe('CopyButton', () => {
  it('should render Share component', () => {
    render(<CopyButton isInTooltip={false} />)
    expect(
      screen.getByRole('button', { name: 'share.svg' })
    ).toBeInTheDocument()
  })

  it('should not render Share component', () => {
    render(<CopyButton isInTooltip={true} />)
    expect(screen.getByRole('button')).toMatchSnapshot()
  })

  it('should render with text', () => {
    render(<CopyButton text={'test'} />)
    expect(screen.getByText('test')).toBeInTheDocument()
  })

  it('should call Fn after click', () => {
    const handleClickMock = jest.fn()

    render(<CopyButton handleClick={handleClickMock} />)

    fireEvent.click(screen.getByRole('button'))

    expect(handleClickMock).toBeCalledTimes(1)
  })
})
