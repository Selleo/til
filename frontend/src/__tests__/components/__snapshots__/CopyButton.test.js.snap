// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyButton should not render Share component 1`] = `
<button
  class="post__share-button"
>
  <span
    class="post__share-text"
  />
</button>
`;
